src/sparse_variational.jl

```diff
@@ -184,68 +130,104 @@ which can be found in closed form.
 variational Gaussian process classification." Artificial Intelligence and
 Statistics. PMLR, 2015.
 """
-function AbstractGPs.posterior(approx::SparseVariationalApproximation{NonCentered})
-    fz = approx.fz
-    data = (Cuu=_chol_cov(fz), C_ε=_chol_cov(approx.q))
-    return ApproxPosteriorGP(approx, fz.f, data)
+function AbstractGPs.posterior(sva::SparseVariationalApproximation{NonCentered})
+    # u = Lk v + mean(fz), v ~ q
+    # m* = K*u Kuu⁻¹ Lk (mean(u) - mean(fz))
+    #    = K*u (Lk Lkᵀ)⁻¹ Lk mean(q)
+    #    = K*u Lk⁻ᵀ Lk⁻¹ Lk mean(q)
+    #    = K*u Lk⁻ᵀ mean(q)
+    #    = K*u α
+    # NonCentered: α = Lk⁻ᵀ m
+    # Centered: α = Kuu⁻¹ (m - mean(fz))
+    # V** = K** - K*u (Kuu⁻¹ - Kuu⁻¹ Lk cov(q) Lkᵀ Kuu⁻¹) Ku*
+    #     = K** - K*u (Kuu⁻¹ - (Lk Lkᵀ)⁻¹ Lk cov(q) Lkᵀ (Lk Lkᵀ)⁻¹) Ku*
+    #     = K** - K*u (Kuu⁻¹ - Lk⁻ᵀ Lk⁻¹ Lk cov(q) Lkᵀ Lk⁻ᵀ Lk⁻¹) Ku*
+    #     = K** - K*u (Kuu⁻¹ - Lk⁻ᵀ cov(q) Lk⁻¹) Ku*
+    #     = K** - (K*u Lk⁻ᵀ) (Lk⁻¹ Ku*) - (K*u Lk⁻ᵀ) Lq Lqᵀ (Lk⁻¹ Ku*)
+    #     = K** - A'A - (K*u Lk⁻ᵀ) Lq Lqᵀ (Lk⁻¹ Ku*)
+    # A = Lk⁻¹ Ku*
+    # NonCentered: B = Lq
+    # Centered: B = Lk⁻¹ Lq
+    q, fz = sva.q, sva.fz
+    m = mean(q)
+    Kuu = _chol_cov(fz)
+    α = chol_lower(Kuu)' \ m
+    Sv = _chol_cov(q)
+    B = chol_lower(Sv)
+    data = (Kuu=Kuu, B=B, α=α)
+    return ApproxPosteriorGP(sva, fz.f, data)
+end
+
+function AbstractGPs.posterior(
+    sva::SparseVariationalApproximation, fx::FiniteGP, ::AbstractVector{<:Real}
+)
+    @assert sva.fz.f === fx.f
+    return posterior(sva)
 end
 
 #
 # Various methods implementing the Internal AbstractGPs API.
 # See AbstractGPs.jl API docs for more info.
 #
 
+function Statistics.mean(
+    f::ApproxPosteriorGP{<:SparseVariationalApproximation}, x::AbstractVector
+)
+    return mean(f.prior, x) + cov(f.prior, x, inducing_points(f)) * f.data.α
+end
+
 # Produces a matrix that is consistently referred to as A in this file. A more descriptive
 # name is, unfortunately, not obvious. It's just an intermediate quantity that happens to
 # get used a lot.
-_A(f, x) = f.data.Cuu.U' \ cov(f.prior, inducing_points(f), x)
-
-function Statistics.mean(
-    f::ApproxPosteriorGP{<:SparseVariationalApproximation{NonCentered}}, x::AbstractVector
-)
-    return mean(f.prior, x) + _A(f, x)' * mean(f.approx.q)
+function _A_and_Cux(f, x)
```

Comment:
At the risk of being a massive pedant, could we please rename `Cux` to `Czx`, i.e. consistently referring to inputs rather than random variables?